primitives/src/date_picker.rs

```diff
@@ -0,0 +1,403 @@
+#[derive(Props, Clone, PartialEq)]
+pub struct DatePickerTriggerProps {
+    /// Additional attributes for the trigger button
+    #[props(extends = GlobalAttributes)]
+    pub attributes: Vec<Attribute>,
+
+    /// The children to render inside the trigger
+    pub children: Element,
+}
+
+/// # DatePickerTrigger
+///
+/// The `PopoverTrigger` is a button that toggles the visibility of the [`PopoverContent`].
+///
+/// ```rust
+/// ```
+#[component]
+pub fn DatePickerTrigger(props: DatePickerTriggerProps) -> Element {
+    let mut ctx = use_context::<DatePickerContext>();
+    let mut open = ctx.open;
+
+    use_effect(move || {
+        let date = (ctx.selected_date)();
+        ctx.set_date(date);
+    });
+
+    rsx! {
+        PopoverRoot {
+            class: "popover",
+            open: open(),
+            on_open_change: move |v| open.set(v),
+            PopoverTrigger { attributes: props.attributes,
+                svg {
+                    class: "date-picker-expand-icon",
+                    view_box: "0 0 24 24",
+                    xmlns: "http://www.w3.org/2000/svg",
+                    polyline { points: "6 9 12 15 18 9" }
+                }
+            }
+            PopoverContent {
+                gap: "0.25rem",
+                align: ContentAlign::End,
+                {props.children}
+            }
+        }
+    }
+}
+
+/// The props for the [`DatePickerInput`] component
+#[derive(Props, Clone, PartialEq)]
+pub struct DatePickerInputProps {
+    /// Additional attributes for the value element
+    #[props(extends = GlobalAttributes)]
+    pub attributes: Vec<Attribute>,
+
+    /// The children of the date picker element
+    pub children: Element,
+}
+
+/// # DatePickerInput
+///
+/// The input element for the [`DatePicker`](super::date_picker::DatePicker) component which allow users to enter a date value.
+///
+/// ```rust
+/// ```
+#[component]
+pub fn DatePickerInput(props: DatePickerInputProps) -> Element {
+    let mut ctx = use_context::<DatePickerContext>();
+
+    let display_value = use_memo(move || ctx.value.to_string());
+
+    let placeholder = {
+        let capacity = (ctx.value)().part_count();
+        let text = ctx.format_placeholder.call(());
+        vec![text; capacity].join(ctx.separator)
+    };
+
+    let handle_input = move |e: Event<FormData>| {
+        let text = e.value().parse().unwrap_or(display_value());
+
+        let value = (ctx.value)();
+        let format = format_description!("[year]-[month]-[day]");
+
+        if value.is_range {
+        } else {
+            if text.is_empty() {
+                ctx.set_date(None);
+                return;
+            }
+
+            let date = Date::parse(&text, &format).ok();
+            if date.is_some_and(|_| !value.is_date_selected(date)) {
+                ctx.set_date(date);
+            }
+        }
+    };
+
+    rsx! {
+        input {
+            style: "min-width: 240px",
```

Comment:
Does this style need to be on the primitives version of the component or could this be in the preview?

Reply:
This property should potentially be dynamic depending on whether we select a date or a range.

In any case, this is necessary to ensure the placeholder is fully displayed on the screen without overlapping, so this setting should be inside the component.

Reply:
The width should be able to change in the user version of the component based on the styling. I agree this is required to use the styled component, but I think it should be in `preview/src/components/date_picker/component.rs` instead of here since it is only related to styling

Keeping the styles in that version of the component makes it possible to change them when you add the component to your project without forking the primitives library

Reply:
After refactoring, this style lost its meaning